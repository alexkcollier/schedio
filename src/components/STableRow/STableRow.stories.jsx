import { SHeading } from '@/components/SHeading'
import { STable } from '@/components/STable'
import { STableCell } from '@/components/STableCell'
import { STableRow } from '@/components/STableRow'
import PropList from '@@/docs/components/PropList'
import { text } from '@storybook/addon-knobs'

export default {
  title: 'Components|Tables/STableRow',

  parameters: {
    jest: 'STableRow',
  },
}

export const tableRow = () => ({
  props: {
    slotContent: {
      default: text('default', 'Slot content', 'Slots'),
    },
  },
  render (h) {
    return (
      <div>
        <SHeading level="1">Table Row</SHeading>
        <p>
          Table rows are containers for <code>{'<STableCell />'}</code>. They do not have any
            props, but have a single default slot. <code>{'<STableCell />'}</code> is the only
          content that should go in this slot.
        </p>

        <SHeading level="2">Example</SHeading>
        <STable>
          <STableRow slot="header">
            <STableCell>Header</STableCell>
          </STableRow>

          <STableRow>{this.slotContent}</STableRow>
        </STable>

        {STableRow.props && <PropList component={STableRow} />}
      </div>
    )
  },
})

tableRow.story = {
  name: 'Table Row',
}
