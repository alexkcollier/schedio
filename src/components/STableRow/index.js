import STableRow from './STableRow.vue';

export { STableRow };

export default { install: Vue => Vue.component(STableRow.name, STableRow) };
