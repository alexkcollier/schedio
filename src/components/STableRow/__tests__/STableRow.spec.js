import { shallowMount } from '@vue/test-utils'
import STableRow from '@/components/STableRow/STableRow.vue'

describe('STableRow.vue', () => {
  it('renders correctly', () => {
    const wrapper = shallowMount(STableRow, {
      slots: {
        default: '<td>Slot content</td>'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
