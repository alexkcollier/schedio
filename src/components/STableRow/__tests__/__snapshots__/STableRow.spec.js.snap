// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`STableRow.vue renders correctly 1`] = `
<tr class="table__row">
  <td>Slot content</td>
</tr>
`;
